src/test/java/tennisclub/dao/UserDaoImplTest.java

```diff
@@ -0,0 +1,170 @@
+package tennisclub.dao;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.test.annotation.DirtiesContext;
+import tennisclub.entity.User;
+import tennisclub.entity.enums.Role;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class UserDaoImplTest {
+    @PersistenceContext
+    EntityManager manager;
+
+    @Autowired
+    private UserDao userDao;
+
+    @Test
+    @Transactional
+    void notNullValidation() {
+        User blankUser = new User();
+        assertThrows(DataIntegrityViolationException.class, () -> userDao.create(blankUser));
+
+        User userWithUsername = new User();
+        userWithUsername.setUsername("username666");
+        assertDoesNotThrow(() -> userDao.create(userWithUsername));
+    }
+
+    @Test
+    @Transactional
+    void checkAllFieldsPersisted() {
+        User createdUser = createUser("honza42", "hon@za.cz", "Honza", Role.USER);
+        manager.persist(createdUser);
+
+        User foundUser = manager.createQuery("select u from User u", User.class).getSingleResult();
```

Comment:
Now, this test doesn't use the Dao at all. I think the user should be persisted using `userDao.create`.

Reply:
@bigno78 Sorry and thank you for noticing. Fixed.